src/src/DataStructs/ESPEasyLimits.h

```diff
@@ -73,15 +125,25 @@
 // * Other operational limits
 // ***********************************************************************
 
-#define MAX_FLASHWRITES_PER_DAY           100 // per 24 hour window
-#define UDP_PACKETSIZE_MAX               2048
-#define TIMER_GRATUITOUS_ARP_MAX           5000
+#ifndef MAX_FLASHWRITES_PER_DAY
+  #define MAX_FLASHWRITES_PER_DAY           100 // per 24 hour window
+#endif
+#ifndef UDP_PACKETSIZE_MAX
+  #define UDP_PACKETSIZE_MAX               2048
+#endif
+#ifndef TIMER_GRATUITOUS_ARP_MAX
+  #define TIMER_GRATUITOUS_ARP_MAX           5000
+#endif
 
-#define UNIT_NUMBER_MAX                  9999  // Stored in Settings.Unit
-#define DOMOTICZ_MAX_IDX            999999999  // Looks like it is an unsigned int, so could be up to 4 bln.
+#ifndef UNIT_NUMBER_MAX
+  #define UNIT_NUMBER_MAX                  9999  // Stored in Settings.Unit
```

Comment:
This is stored (and used) as a byte, so do not go above 254.
Unit 255 is used as a broadcast ID in the p2p protocol.

Reply:
I did not change this!! I only added the `#ifndef` above, so this `UNIT_NUMBER_MAX` already was at 9999 before!

Reply:
what's the difference between `UNIT_MAX` and `UNIT_NUMBER_MAX`?

I think `UNIT_MAX` is used to iterate over the unit's in `WebServer_CustomPage.ino` and `UNIT_NUMBER_MAX` is used in the same file but for the Textbox. Eventually they should be the same?

Reply:
Ik think they indeed need to be the same

Reply:
Maybe there used to be a difference once when the limit was set to 31 nodes?

Reply:
Could be, but as `UNIT_NUMBER_MAX` is only used to check the validity of the Textbox and not really in the code, I think this is just another cleanup. Throw `UNIT_NUMBER_MAX` out in `WebServer_CustomPage.ino` and replace it with `UNIT_MAX` so it's consistent with what's used in the code for iterating... otherwise a user could set a number bigger than the 254 `defined in UNIX_MAX`.
Probably the difference is, that while you can have a maximum of `UNIT_MAX` nodes, you can give them individual numbers up to `UNIT_NUMBER_MAX`. There fore you can have a node that's numbered 1234 but still stay below the 254 nodes maximum....  but I would need to further dig into the code to be sure

Reply:
I already merged them in my code, will commit to my active PR in a minute.

Reply:
so you think there should be no difference between number of nodes and node-number?

Reply:
Pushed the commit. I think you should base your search on that PR, since I also incorporated a lot of your changes too.

Reply:
> so you think there should be no difference between number of nodes and node-number?

It was about the max of the unit number.
And I think it should be the same max indeed.
We are storing it in a byte and also use it in other places as a byte (e.g. p2p protocol), so we cannot increase its value beyond that.

Reply:
Agree...

BTW: which PR are you referring / preparing? can't seem to find it...

I'll try later on with that... I'm offline for the next hour or so ;) need to do a bit of "real" work :D

Reply:
https://github.com/letscontrolit/ESPEasy/pull/2617

Reply:
> #2617

ah... Title is a bit misleading... a lot in one PR ;) I'll test it later...

Reply:
in `src/DataStructs/Caches.h`you missed one include and need to change
```
#include "ESPEasy_common.h"
```
to
```
#include "../../ESPEasy_common.h"
```
to make it compile in ArduinoIDE

Reply:
Done

Reply:
ok, tried it, where should I post findings? here or in the newly opened 24-tasks discussion?

Reply:
Depends if they are related to that.
If they are related to Arduino IDE builds, I would suggest to start a new issue for that since that part is also becoming a real issue right now...

Reply:
arduino builds again fine with this PR. so I guess I put it to the 24 tasks issue, as it really happens when (self) defining 24 tasks after the cleanup merge..